src/deploy.js

```diff
@@ -1,51 +1,109 @@
 import AWS from 'aws-sdk';
 import s3 from 's3-node-client';
 import dotenv from 'dotenv';
+import fs from 'fs';
+import cliProgress from 'cli-progress';
 
-const path = require('path');
+const validateTag = (tag) => {
+  // Both compilation hints because of backslashes used in RegExp but unecessary by conception in JS Strings
+  // prettier-ignore
+  // eslint-disable-next-line no-useless-escape
+  const pattern = new RegExp('v\\d+(\.\\d+){0,2}$');
+  if (tag === 'latest' || pattern.test(tag)) {
+    console.log(`info: validated tag ${tag}`);
+    return true;
+  }
+  console.error(`error: unable to validate version '${tag}'`);
+  return false;
+};
 
-// default build directory
-const BUILD = 'build/';
+const validateEnv = (env) => {
+  if (fs.existsSync(env)) {
+    console.log(`info: validated environment file ${env}`);
+    return true;
+  }
+  console.log(`error: environment file '${env}' does not exist`);
+  return false;
+};
 
-/**
- * Returns an object with all variables loaded from a environment
- * @param {string} environmentName is the suffix after .env.*
- */
+const validateBuild = (build) => {
+  if (fs.existsSync(build)) {
+    console.log(`info: validated build directory ${build}`);
+    return true;
+  }
+  console.log(`error: build directory '${build}' does not exist`);
+  return false;
+};
 
-const deploy = async (opts) => {
-  // const { path } = opts;
+const isDefined = (variable) => {
+  return typeof variable !== 'undefined';
+};
 
-  const usageMessage = console.log(
-    'usage: $0 [-e <path/to/file>] [-v <version string>] [-b <path/to/build>]',
-  );
+const deploy = async (opts) => {
+  const { tag, env, build } = opts;
 
-  console.log(`Exectued with path: ${opts.path}`);
-  console.log(usageMessage);
+  // Validate command options
+  if (!validateTag(tag) || !validateEnv(env) || !validateBuild(build)) {
+    console.error('Abort...');
+    return false;
+  }
 
   // fetch environment variables
-  dotenv.config({ path: path.resolve(process.cwd(), '.env.dev') });
-  /* eslint-disable no-unused-vars */
+  // dotenv.config({ path: path.resolve(process.cwd(), '.env.dev') });
+  dotenv.config({ path: env });
   const {
     REACT_APP_GRAASP_DEVELOPER_ID,
     REACT_APP_GRAASP_APP_ID,
-    REACT_APP_GRAASP_DOMAIN,
     REACT_APP_HOST,
     REACT_APP_VERSION,
-    REACT_APP_BASE,
-    NODE_ENV,
     BUCKET,
-    AWS_DEFAULT_REGION,
     AWS_ACCESS_KEY_ID,
     AWS_SECRET_ACCESS_KEY,
     DISTRIBUTION,
   } = process.env;
-  /* eslint-enable no-unused-vars */
 
-  AWS.config.getCredentials(function (err) {
-    if (err) console.error(err.stack);
-    // credentials not loaded
-    else {
-      console.log('Access key:', AWS.config.credentials.accessKeyId);
+  // ensure the correct app variables are defined
+  if (
+    !isDefined(REACT_APP_HOST) ||
+    !isDefined(REACT_APP_GRAASP_DEVELOPER_ID) ||
+    !isDefined(REACT_APP_GRAASP_APP_ID)
+  ) {
+    console.error(
```

Comment:
No need to separate into three messages. Just one `console.error`. And you can remove the `error:` bit, which was for the bash script.

Reply:
@ugGit, you forgot to remove the `error:` bit.